conans/client/build/cmake_flags.py

```diff
@@ -18,11 +18,19 @@
 cmake_in_local_cache_var_name = "CONAN_IN_LOCAL_CACHE"
 
 
-def get_toolset(settings):
-    if settings.get_safe("compiler") == "Visual Studio":
+def get_toolset(settings, generator):
+    compiler = settings.get_safe("compiler")
+    compiler_base = settings.get_safe("compiler.base")
+    if compiler == "Visual Studio":
         subs_toolset = settings.get_safe("compiler.toolset")
         if subs_toolset:
             return subs_toolset
+    elif compiler == "intel" and compiler_base == "Visual Studio" and "Visual" in generator:
+        compiler_version = settings.get_safe("compiler.version")
+        if compiler_version:
+            compiler_version = compiler_version if "." in compiler_version else \
+                "%s.0" % compiler_version
+            return "Intel C++ Compiler " + compiler_version
```

Comment:
This still fails with the following profile
```
[settings]
arch=x86_64
arch_build=x86_64
os=Windows
os_build=Windows
compiler=Visual Studio
compiler.version=16
compiler.runtime=MDd
compiler.toolset=v142
build_type=Debug
[env]
CONAN_CMAKE_GENERATOR=Ninja
```
Granted, it also failed before this PR.

Reply:
okay, it was designed to raise on purpose (to detect incompatible config): https://github.com/conan-io/conan/pull/5201
I think I'll leave this specific case as it was before